app/controllers/line_items_controller.rb

```diff
@@ -1,2 +1,52 @@
 class LineItemsController < ApplicationController
+  before_action :find_item, only: %i[ destroy add_quantity reduce_quantity]
+
+  def create       
+    chosen_product = Product.find(params[:product_id])
+    current_cart = @current_cart
+    if current_cart.products.include?(chosen_product)
+      @line_item = current_cart.line_items.find_by(:product_id => chosen_product)
+      @line_item.quantity += 1
+    else
+      @line_item = LineItem.new
+      @line_item.cart = current_cart
+      @line_item.product = chosen_product
+      @line_item.quantity = 1      
+    end
+    @line_item.save!
+    redirect_to cart_path(current_cart)
+  end
+    
+  def destroy
+    @line_item.destroy
+    redirect_to cart_path(@current_cart)
+  end  
+    
+  def add_quantity
+    @line_item.quantity ||= 0
+    @line_item.quantity += 1
+    @line_item.save!
+    redirect_to cart_path(@current_cart)
+  end
+      
+  def reduce_quantity
+    if @line_item.quantity > 1
+      @line_item.quantity -= 1
+    end
+    @line_item.save
```

Comment:
use update_attribute instead of save

Reply:
used update_attribute for reducing quantity

Reply:
I don't see above change